adr/2022_02_11_declaring_services_in_container.md

```diff
@@ -0,0 +1,65 @@
+# Declaring services in container
+
+* Status: accepted
+* Date: 2022-02-11
+
+## Context and Problem Statement
+
+Since Symfony 3.4, all Symfony services are declared private by default. And this is the recommended Symfony approach. On the 
+other hand, first stable Sylius release was published before that and therefore at the moment of publication of this recommendation
+Sylius had thousands (or at least hundreds) public services. The question is how should we handle services declared after the 
+3.4 release.
+
+## Decision Drivers
+
+Declared services should be:
+* services should be easily usable in ResourceControllers (where we are allowing for fetching services directly from container)
+* services should be easily usable in state machine callback (where we are allowing for fetching services directly from container)
```

Comment:
```suggestion
* easily usable in the state machine callback (where we are allowing for fetching services directly from the container)
```

Reply:
Additionally, do we "allow" to fetch services from container used as SM callbacks? I believe it's required by state machine library itself

Reply:
The above change was done. According to the second question: absolutely - you may declare services private and therefore you will block usage in SM. We may discuss if other services that are already used in Sylius should be easily pluggable to SM if you wish. But by default, we may allow or block access to given services from SM callbacks